e2e/testcases/managed_resources_test.go

```diff
@@ -69,19 +72,20 @@ func TestKubectlCreatesManagedNamespaceResourceMultiRepo(t *testing.T) {
 	}
 
 	/* A new test */
-	ns := []byte(`
-apiVersion: v1
-kind: Namespace
-metadata:
-  name: test-ns1
-  annotations:
-    configmanagement.gke.io/managed: enabled
-    configsync.gke.io/resource-id: _namespace_test-ns1
-    configsync.gke.io/manager: :root
-`)
+	ns1Obj := &unstructured.Unstructured{}
+	ns1Obj.SetGroupVersionKind(kinds.Namespace())
+	ns1Obj.SetName("test-ns1")
+	ns1Obj.SetAnnotations(map[string]string{
+		metadata.ResourceManagementKey: metadata.ResourceManagementEnabled,
+		metadata.ResourceIDKey:         "_namespace_test-ns1",
+		metadata.ResourceManagerKey:    rootSync1Manager,
+	})
+	ns1Obj.SetLabels(map[string]string{
+		metadata.ApplySetPartOfLabel: rootSync1ApplySetID,
+	})
 
-	if err := os.WriteFile(filepath.Join(nt.TmpDir, "test-ns1.yaml"), ns, 0644); err != nil {
-		nt.T.Fatalf("failed to create a tmp file %v", err)
+	if err := writeObjectYAMLFile(filepath.Join(nt.TmpDir, "test-ns1.yaml"), ns1Obj, nt.Scheme); err != nil {
```

Comment:
Is there a reason we need to write yaml to the filesystem instead of applying with the KubeClient?

Reply:
The test was explicitly applying with kubectl already, I think to simulate using another client more fully as an example.

I only did a quick change to make it easier to use the existing constants. 

But no, I don't think we really need to use kubectl. You wanna rewrite the test to simplify? Seemed out of scope for this PR.

Reply:
Yeah that's fine, we can refactor in a follow-up

Reply:
Looks like all the tests in this file use kubectl... Maybe that's better than using client-go directly because it more accurately reflects the type of change the user would want reverted, as opposed to one from a controller.

kubectl also might add annotations too, tho we don't test that they are preserved or reverted.